jsonschema-module-microprofile-openapi-3/src/main/java/com/github/victools/jsonschema/module/microprofile/openapi3/MicroProfileOpenApi3Module.java

```diff
@@ -0,0 +1,553 @@
+        Optional<String> externalReference = this.getSchemaAnnotationValue(scope, Schema::ref, ref -> !ref.isEmpty());
+        if (!externalReference.isPresent()) {
+            return null;
+        }
+        // in Draft 6 and Draft 7, no other keywords are allowed besides a "$ref"
+        CustomDefinition.AttributeInclusion attributeInclusion;
+        switch (context.getGeneratorConfig().getSchemaVersion()) {
+        case DRAFT_6:
+        // fall-through (same as Draft 7)
+        case DRAFT_7:
+            attributeInclusion = CustomDefinition.AttributeInclusion.NO;
+            break;
+        default:
+            attributeInclusion = CustomDefinition.AttributeInclusion.YES;
+        }
+        ObjectNode reference = context.getGeneratorConfig().createObjectNode()
+                .put(context.getKeyword(SchemaKeyword.TAG_REF), externalReference.get());
+        return new CustomPropertyDefinition(reference, attributeInclusion);
+    }
+
+    /**
+     * Consider various remaining aspects.
+     * <ul>
+     * <li>{@code @Schema(not = ...)}</li>
+     * <li>{@code @Schema(allOf = ...)}</li>
+     * <li>{@code @Schema(minProperties = ...)}</li>
+     * <li>{@code @Schema(maxProperties = ...)}</li>
+     * <li>{@code @Schema(requiredProperties = ...)}</li>
+     * </ul>
+     *
+     * @param memberAttributes already collected schema for the field/method
+     * @param member           targeted field/method
+     * @param context          generation context
+     */
+    protected void overrideInstanceAttributes(ObjectNode memberAttributes, MemberScope<?, ?> member, SchemaGenerationContext context) {
+        Schema annotation = this.getSchemaAnnotationValue(member, Function.identity(), x -> true)
+                .orElse(null);
+        if (annotation == null) {
+            return;
+        }
+        if (annotation.not() != Void.class) {
+            memberAttributes.set(context.getKeyword(SchemaKeyword.TAG_NOT),
+                    context.createDefinitionReference(context.getTypeContext().resolve(annotation.not())));
+        }
+        if (annotation.allOf().length > 0) {
+            ArrayNode allOfArray = memberAttributes.withArray(context.getKeyword(SchemaKeyword.TAG_ALLOF));
+            Stream.of(annotation.allOf())
+                    .map(context.getTypeContext()::resolve)
+                    .map(context::createDefinitionReference)
+                    .forEach(allOfArray::add);
+        }
+        if (annotation.anyOf().length > 0) {
+            ArrayNode allOfArray = memberAttributes.withArray(context.getKeyword(SchemaKeyword.TAG_ALLOF));
+            ArrayNode anyOfArray = allOfArray.addObject().withArray(context.getKeyword(SchemaKeyword.TAG_ANYOF));
+            Stream.of(annotation.anyOf())
+                    .map(context.getTypeContext()::resolve)
+                    .map(context::createDefinitionReference)
+                    .forEach(anyOfArray::add);
+        }
```

Comment:
**suggestion**: As per my other comment, I suggest letting the `anyOf` options be handled as subtypes. This may resolut in a cleaner schema being generated.

Reply:
By removing this bloc and keep `MicroProfileOpenApi3AnyOfResolver` my test on `Foo.class` is failing.